test/utils.py

```diff
@@ -665,6 +665,13 @@ def joinall(threads):
         assert not t.is_alive(), "Thread %s hung" % t
 
 
+async def asyncjoinall(tasks):
+    """Join tasks with a 5-minute timeout, assert joins succeeded"""
+    for t in tasks:
+        await asyncio.wait_for(t.task, 300)
```

Comment:
```suggestion
        await asyncio.wait([t.task], timeout=300)
```

`wait_for` will throw a `TimeoutError` if it times out, `wait` will just return the task that didn't complete.

Reply:
ahh i didn't know that!